cli/cmd/contracts.go

```diff
@@ -94,6 +95,84 @@ var (
 		},
 	}
 
+	contractsStorageCmd = &cobra.Command{
+		Use:   "storage",
+		Short: "WebAssembly smart contracts storage operations",
+	}
+
+	contractsStorageDumpCmd = &cobra.Command{
+		Use:   "dump <instance-id>",
+		Short: "Dump instantiated public contract state in JSON",
+		Args:  cobra.ExactArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			cfg := cliConfig.Global()
+			npa := common.GetNPASelection(cfg)
+			strInstanceID := args[0]
+
+			if npa.ParaTime == nil {
+				cobra.CheckErr("no paratimes configured")
+			}
+
+			instanceID, err := strconv.ParseUint(strInstanceID, 10, 64)
+			cobra.CheckErr(err)
+
+			ctx := context.Background()
+			conn, err := connection.Connect(ctx, npa.Network)
+			cobra.CheckErr(err)
+
+			res, err := conn.Runtime(npa.ParaTime).Contracts.InstanceRawStorage(ctx, client.RoundLatest, contracts.InstanceID(instanceID))
+			cobra.CheckErr(err)
+
+			storage := map[string]interface{}{}
+			for _, k := range res.Value {
+				var val interface{}
+				err = cbor.Unmarshal(k.Value, &val)
+				cobra.CheckErr(err)
+				storage[fmt.Sprintf("%s", k.Key)] = val
+			}
+
+			storageJSON, err := json.Marshal(storage)
+			cobra.CheckErr(err)
+
+			fmt.Printf("%s\n", storageJSON)
+		},
+	}
+
+	contractsStorageGetCmd = &cobra.Command{
+		Use:   "get <instance-id> <key>",
+		Short: "Fetch value of given key from public contract state",
+		Args:  cobra.ExactArgs(2),
+		Run: func(cmd *cobra.Command, args []string) {
+			cfg := cliConfig.Global()
+			npa := common.GetNPASelection(cfg)
+			strInstanceID := args[0]
+			key := args[1]
+
+			if npa.ParaTime == nil {
+				cobra.CheckErr("no paratimes configured")
+			}
+
+			instanceID, err := strconv.ParseUint(strInstanceID, 10, 64)
+			cobra.CheckErr(err)
+
+			ctx := context.Background()
+			conn, err := connection.Connect(ctx, npa.Network)
+			cobra.CheckErr(err)
+
+			res, err := conn.Runtime(npa.ParaTime).Contracts.InstanceStorage(ctx, client.RoundLatest, contracts.InstanceID(instanceID), []byte(key))
+			cobra.CheckErr(err)
+
+			var storageCell interface{}
+			err = cbor.Unmarshal(res.Value, &storageCell)
+			cobra.CheckErr(err)
```

Comment:
As above, this should be made more robust with fallback to Base64 encoding of the raw binary value in case CBOR decoding fails.